marketing/views.py

```diff
@@ -15,9 +18,53 @@ class QuoteListView(generics.ListAPIView):
 class QuoteCreateView(generics.CreateAPIView):
     serializer_class = QuoteSerializer
 
+    def create(self, request, *args, **kwargs):
+        program_id = request.headers.get('Program-ID')
+        quote_serializer = self.get_serializer(data=request.data)
+        quote_serializer.is_valid(raise_exception=True)
+        quote = quote_serializer.save(author=request.user)
+
+        try:
+            program = Program.objects.get(id=program_id)
+            program.quotes.add(quote)
+            return Response({"message": "Quote attached successfully"})
+        except Program.DoesNotExist:
+            return Response({"message": "Program not found"})
+
 class QuoteRetrieveUpdateDestroyView(generics.RetrieveUpdateDestroyAPIView):
     queryset = Quote.objects.all()
     lookup_field = 'id'
     serializer_class = QuoteSerializer
     parser_classes = (MultiPartJSONParser,JSONParser,)
 
+    def patch(self, request, *args, **kwargs):
+        program_id = request.headers.get('Program-ID')
+        attribute = request.headers.get('Attr-Quote')
```

Comment:
See note above about trying to replace these headers by passing the information in the request body.

You can get the data in the request using `request.data`, which is a dictionary.